### DISCUSSION TITLE
Does Langsmith logs Ingestions basically embedding logs(recursive character text splitter, vectordb ingestion traces  etc)

### DISCUSSION THREAD

**Discussion by akash97715** [langchain-ai/langsmith-sdk#1109] [Q&A]
I am trying to use langsmith as product for my project, thinking of enterprise one. One doubt does langsmith logs the embedding ingestion logs(Recursive charctertext splitter, vectordb storing piece)

**Comment by segfly** [comment]
I have a similar need, starting with tracking embedding API costs. Unlike the LLM/chat models, it does not appear that "langchain-provided" embedding models are integrated yet with langsmith (or maybe modules like `langchain_openai` are 3rd party maintained, and the maintainer hasn't done it yet - I don't know). I've been scouring the docs but can't find any mention of tracing embedding calls and it doesn't show up in langsmith traces out of the box.

It appears one has to [annotate your own runnable/graphs](https://docs.smith.langchain.com/how_to_guides/tracing/log_llm_trace#manually-provide-token-counts) to provide the metadata you wish (like tokens) and then [add to langsmith settings your provider/model cost](https://docs.smith.langchain.com/how_to_guides/tracing/calculate_token_based_costs).

The tracing facility seems quite flexible. You can add traces to anything you wish and abuse the langsmith custom model to represent any kind of transaction costs (DB calls, API calls, embedding tokens, etc.).
